src/coverlet.core/Abstracts/IInstrumenter.cs

```diff
@@ -0,0 +1,126 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Globalization;
+using System.IO;
+using Newtonsoft.Json;
+
+namespace Coverlet.Core
+{
+    public class Line
+    {
+        public int Number;
+        public string Class;
+        public string Method;
+        public int Hits;
+    }
+
+    public class Branch : Line
+    {
+        public int Offset;
+        public int EndOffset;
+        public int Path;
+        public uint Ordinal;
+    }
+
+    public class BranchKeyConverter : TypeConverter
```

Comment:
we need custom converter because Json serializer does not support serialization of complex dictionary key